vpc/vpc-private-subnet.template

```diff
@@ -0,0 +1,658 @@
+{
+  "AWSTemplateFormatVersion": "2010-09-09",
+  "Description": "3 private subnets and 3 NATs across 3 availability zones in a VPC",
+  "Parameters": {
+    "VpcId": {
+      "Description": "The VPC ID we are deploying assets into.",
+      "Type": "AWS::EC2::VPC::Id"
+    },
+    "ServiceName": {
+      "Description": "Name of this deployment",
+      "AllowedPattern": "^[a-z0-9][a-z0-9-.]*$",
+      "Type": "String"
+    },
+    "TechnicalOwner": {
+      "Description": "A valid LDAP email",
+      "AllowedPattern": "^([a-zA-Z0-9_\\-\\.]+)@((\\[[0-9]{1,3}\\.[0-9]{1,3}\\.[0-9]{1,3}\\.)|(([a-zA-Z0-9\\-]+\\.)+))([a-zA-Z]{2,4}|[0-9]{1,3})(\\]?)$",
+      "Type": "String"
+    },
+    "Environment": {
+      "Description": "The name of the environment we are deploying into: sandbox, stage, or production.",
+      "Type": "String",
+      "AllowedValues": [
+        "sandbox",
+        "stage",
+        "prod",
+        "admin"
+      ],
+      "Default": "sandbox",
+      "ConstraintDescription": "Must be one of these types of VPCs"
+    },
+    "PrivateSubnetAZ1Cidr": {
+      "Description": "The CIDR for the Private Subnet in Availability Zone 1.",
+      "Type": "String",
+      "MinLength": "9",
+      "MaxLength": "19",
+      "AllowedPattern": "^([0-9]|[1-9][0-9]|1[0-9]{2}|2([0-4][0-9]|5[0-5])).([0-9]|[1-9][0-9]|1[0-9]{2}|2([0-4][0-9]|5[0-5])).([0-9]|[1-9][0-9]|1[0-9]{2}|2([0-4][0-9]|5[0-5])).([0-9]|[1-9][0-9]|1[0-9]{2}|2([0-4][0-9]|5[0-5]))/([0-9]|[12][0-9]|3[0-2])",
+      "ConstraintDescription": "Must be a valid IP CIDR range of the form x.x.x.x/x"
+    },
+    "PrivateSubnetAZ2Cidr": {
+      "Description": "The CIDR for the Private Subnet in Availability Zone 2.",
+      "Type": "String",
+      "MinLength": "9",
+      "MaxLength": "19",
+      "AllowedPattern": "^([0-9]|[1-9][0-9]|1[0-9]{2}|2([0-4][0-9]|5[0-5])).([0-9]|[1-9][0-9]|1[0-9]{2}|2([0-4][0-9]|5[0-5])).([0-9]|[1-9][0-9]|1[0-9]{2}|2([0-4][0-9]|5[0-5])).([0-9]|[1-9][0-9]|1[0-9]{2}|2([0-4][0-9]|5[0-5]))/([0-9]|[12][0-9]|3[0-2])",
+      "ConstraintDescription": "Must be a valid IP CIDR range of the form x.x.x.x/x"
+    },
+    "PrivateSubnetAZ3Cidr": {
+      "Description": "The CIDR for the Private Subnet in Availability Zone 3.",
+      "Type": "String",
+      "MinLength": "9",
+      "MaxLength": "19",
+      "AllowedPattern": "^([0-9]|[1-9][0-9]|1[0-9]{2}|2([0-4][0-9]|5[0-5])).([0-9]|[1-9][0-9]|1[0-9]{2}|2([0-4][0-9]|5[0-5])).([0-9]|[1-9][0-9]|1[0-9]{2}|2([0-4][0-9]|5[0-5])).([0-9]|[1-9][0-9]|1[0-9]{2}|2([0-4][0-9]|5[0-5]))/([0-9]|[12][0-9]|3[0-2])",
+      "ConstraintDescription": "Must be a valid IP CIDR range of the form x.x.x.x/x"
+    },
+    "SSHKeyName": {
+      "Description": "Name of an existing EC2 SSH KeyPair to install on the instance",
```

Comment:
Yes. This is one thing I am not happy with in this template. This one parameter is required even if the nat (private subnet stack) is not required. The issue is that I can either use the native type to validate and require the parameter or not require the parameter (set a bogus default) and then not validate the input.